src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs

```diff
@@ -197,11 +197,19 @@ private void ValidateTokens(HttpContext httpContext, AntiforgeryTokenSet antifor
             // Extract cookie & request tokens
             AntiforgeryToken deserializedCookieToken;
             AntiforgeryToken deserializedRequestToken;
-            DeserializeTokens(
-                httpContext,
-                antiforgeryTokenSet,
-                out deserializedCookieToken,
-                out deserializedRequestToken);
+
+            try
+            {
+                DeserializeTokens(
+                    httpContext,
+                    antiforgeryTokenSet,
+                    out deserializedCookieToken,
+                    out deserializedRequestToken);
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new AntiforgeryValidationException(Resources.Antiforgery_FailedToDeserializeTokens, ex);
+            }
```

Comment:
I think this needs a slightly different solution, especially since we own the token serializer code. Putting a big try/catch with a general exception type like `IOE` around this much code seem sloppy 👎 I'd much rather see this inside the token serializer - which I believe is not extensible.

We also need to consider the other call to `DeserializeTokens` inside `TryValidate...` that needs to be updated as well.